pkg/inputs/mysqlbatch/input.go

```diff
@@ -322,16 +322,16 @@ func (plugin *mysqlBatchInputPlugin) waitFinish(positionCache position_store.Pos
 		}
 
 		binlogPositionsValue := helper.BinlogPositionsValue{
-			StartPosition:   startBinlog,
-			CurrentPosition: startBinlog,
+			StartPosition:   &startBinlog,
+			CurrentPosition: &startBinlog,
```

Comment:
this is dangerous. if one update the content of `CurrentPosition`, `StartPosition` will also be updated. a deep copy is better.